diesel/src/query_builder/debug_binds.rs

```diff
@@ -0,0 +1,42 @@
+use std::fmt::{self, Display};
+use std::marker::PhantomData;
+use std::mem;
+
+use backend::Backend;
+use super::{QueryFragment, AstPass};
+
+#[allow(missing_debug_implementations)]
+/// A struct that implements `fmt::Display` by walking the given AST and
+/// writing the `fmt::Debug` implementation of each bind parameter.
+pub struct DebugBinds<'a, T: 'a, DB> {
+    query: &'a T,
+    _marker: PhantomData<DB>,
+}
+
+impl<'a, T, DB> DebugBinds<'a, T, DB> {
+    pub fn new(query: &'a T) -> Self {
+        DebugBinds {
+            query,
+            _marker: PhantomData,
+        }
+    }
+}
+
+impl<'a, T, DB> Display for DebugBinds<'a, T, DB> where
+    DB: Backend,
+    T: QueryFragment<DB>,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str("binds: ")?;
```

Comment:
I was hoping to keep them semantically separate since if we go with `slog` we'll actually want to keep them separate in that structure. If we did want to do the json-ish thing, perhaps instead of a `debug_sql` function/macro, we should have a `.debug()` instance method which just returns a struct that implements `std::fmt::Debug`